modules/nf-core/openms/idfilter/main.nf

```diff
@@ -0,0 +1,57 @@
+process OPENMS_IDFILTER {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::openms=3.1.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_2' :
+        'biocontainers/openms:3.1.0--h8964181_2' }"
+
+    input:
+    tuple val(meta), path(id_file), path(filter_file)
+
+    output:
+    tuple val(meta), path("*.{idXML,consensusXML}"), emit: filtered
+    path "versions.yml"                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def suffix = task.ext.suffix ?: "${id_file.getExtension()}"
+    // Optional filtering via filter_file
+    def filter_citerion = task.ext.args2 ?: "-whitelist:peptides"
+    def filter = filter_file ? "${filter_citerion} ${filter_file}" : ""
```

Comment:
```suggestion
    def filter = filter_file ? "-whitelist:peptides ${filter_file}" : ""
```

Reply:
There can be multiple filter criteria, e.g. `-whitelist:peptides|proteins`, `-blacklist:peptides|proteins` `-in_silico_digestion:fasta`. I guess this needs to be defined in ` task.ext.args2` 🤔 or how would you do it?

Reply:
See [doc](https://abibuilder.cs.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_IDFilter.html)

Reply:
ah ok then it makes sense :)